libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Maxell.h

```diff
@@ -15,6 +15,10 @@ class AP_BattMonitor_SMBus_Maxell : public AP_BattMonitor_SMBus
                              AP_BattMonitor::BattMonitor_State &mon_state,
                              AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev);
 
+    // generic SMBus batteries may not always report the cell voltages, as we
+    // the data is being read from manufacturer specific fields
```

Comment:
I don't think that's true (I'm happy to rename the drivers to be more clear).

The "Maxell" driver only implements SMBus Battery Data Specifications with the one add on of battery cell voltages being read from a combination of OptionalMfGFunctions and reserved fields. However this one add on is also commonly used by other manufacturers (Tattu among others), but for other people like TI chips it has a very different meaning. Now there is an argument we should have a parameter that lets users not query this, but this is the only SMBus driver we have as an option for people if they aren't using a solo.

The solo does current and cell voltages in it's own non standard way which makes it useless for any other smart battery.